### ISSUE TITLE
Incorrect _revisions after replication

### ISSUE THREAD

**Issue by filionf** [pouchdb/pouchdb#5883]
### Issue
After replicating a specific document, it gets an incorrects _revisions sequence. For example, I could have a rev Id starting with 6-, but have only 5 revision ids.

I have a complex environment that automatically merges documents together when a conflict occurs. It also reuse some revhashes in the process, which I think is causing the issue. I do have the response from the server when doing the replication and I believe the answer is somewhere in there.

If you notice, there are multiple open revs and the revision hash 0a21b4bd4399b51e144a06b126031edc exists at 2 places: 5-0a21b4bd4399b51e144a06b126031edc and 6-0a21b4bd4399b51e144a06b126031edc.

Once replicated, retrieving the document using the local database give the following:

`{
  "$createdBy": "a576775bb8378240d20502932f03e14a",
  "$creationDate": "2016-11-17T03:17:32.637Z",
  "status": 1,
  "$category": "ACCIDENT",
  "$lastModifiedUser": "a576775bb8378240d20502932f03e14a",
  "$lastModifiedDate": "2016-11-17T03:17:36.553Z",
  "$timezoneOffset": 300,
  "_id": "1109B3A7-E112-845B-BB39-F530CD2E158A",
  "_rev": "6-0a21b4bd4399b51e144a06b126031edc",
  "_revisions": {
    "start": 5,
    "ids": [
      "0a21b4bd4399b51e144a06b126031edc",
      "90f7c5665484a4972e8eb3c5baee6b15",
      "aa8f5ef9544876ec652bf21b500cb890",
      "57152ea3e06e0e6580941ed69c2d0729",
      "a44a851eb78ed461b6051b09fdae0813"
    ]
  }
}`

Any hint on how I could fix this would be appreciated.

### Info
- Environment: Node.js Server / Browser Client
- Platform: Chrome
- Adapter: WebSQL
- Server: CouchDB

### Reproduce

GET $couchdb/1109B3A7-E112-845B-BB39-F530CD2E158A?revs=true&open_revs=%5B%226-0a21b4bd4399b51e144a06b126031edc%22%2C%226-b424cf4593313b08af3757097e9f79dc%22%2C%225-98bba7dd19fc74bd519f0ce8962a3609%22%2C%224-bdd85d57bc1e52e7074a7c79d97182d8%22%5D

`[{
	"ok": {
		"_id": "1109B3A7-E112-845B-BB39-F530CD2E158A",
		"_rev": "4-bdd85d57bc1e52e7074a7c79d97182d8",
		"$createdBy": "a576775bb8378240d20502932f03e14a",
		"$creationDate": "2016-11-17T03:17:32.637Z",
		"status": 1,
		"id": 869863,
		"$category": "ACCIDENT",
		"$lastModifiedUser": "a576775bb8378240d20502932f03e14a",
		"$lastModifiedDate": "2016-11-17T03:17:36.553Z",
		"$timezoneOffset": 300,
		"_deleted": true,
		"_revisions": {
			"start": 4,
			"ids": ["bdd85d57bc1e52e7074a7c79d97182d8", "90f7c5665484a4972e8eb3c5baee6b15", "57152ea3e06e0e6580941ed69c2d0729", "a44a851eb78ed461b6051b09fdae0813"]
		}
	}
}, {
	"ok": {
		"_id": "1109B3A7-E112-845B-BB39-F530CD2E158A",
		"_rev": "5-98bba7dd19fc74bd519f0ce8962a3609",
		"$createdBy": "a576775bb8378240d20502932f03e14a",
		"$creationDate": "2016-11-17T03:17:32.637Z",
		"status": 1,
		"id": 869863,
		"$category": "ACCIDENT",
		"$lastModifiedUser": "a576775bb8378240d20502932f03e14a",
		"$lastModifiedDate": "2016-11-17T03:17:36.553Z",
		"$timezoneOffset": 300,
		"_deleted": true,
		"_revisions": {
			"start": 5,
			"ids": ["98bba7dd19fc74bd519f0ce8962a3609", "5cecb8ebfd59bff046d889b0096646e6", "aa8f5ef9544876ec652bf21b500cb890", "57152ea3e06e0e6580941ed69c2d0729", "a44a851eb78ed461b6051b09fdae0813"]
		}
	}
}, {
	"ok": {
		"_id": "1109B3A7-E112-845B-BB39-F530CD2E158A",
		"_rev": "6-b424cf4593313b08af3757097e9f79dc",
		"$createdBy": "a576775bb8378240d20502932f03e14a",
		"$creationDate": "2016-11-17T03:17:32.637Z",
		"status": 1,
		"id": 869863,
		"$category": "ACCIDENT",
		"$lastModifiedUser": "a576775bb8378240d20502932f03e14a",
		"$lastModifiedDate": "2016-11-17T03:17:36.553Z",
		"$timezoneOffset": 300,
		"_deleted": true,
		"_revisions": {
			"start": 6,
			"ids": ["b424cf4593313b08af3757097e9f79dc", "0a21b4bd4399b51e144a06b126031edc", "90f7c5665484a4972e8eb3c5baee6b15", "aa8f5ef9544876ec652bf21b500cb890", "57152ea3e06e0e6580941ed69c2d0729", "a44a851eb78ed461b6051b09fdae0813"]
		}
	}
}, {
	"ok": {
		"_id": "1109B3A7-E112-845B-BB39-F530CD2E158A",
		"_rev": "6-0a21b4bd4399b51e144a06b126031edc",
		"$createdBy": "a576775bb8378240d20502932f03e14a",
		"$creationDate": "2016-11-17T03:17:32.637Z",
		"status": 1,
		"id": 869863,
		"$category": "ACCIDENT",
		"$lastModifiedUser": "a576775bb8378240d20502932f03e14a",
		"$lastModifiedDate": "2016-11-17T03:17:36.553Z",
		"$timezoneOffset": 300,
		"_revisions": {
			"start": 6,
			"ids": ["0a21b4bd4399b51e144a06b126031edc", "5cecb8ebfd59bff046d889b0096646e6", "90f7c5665484a4972e8eb3c5baee6b15", "aa8f5ef9544876ec652bf21b500cb890", "57152ea3e06e0e6580941ed69c2d0729", "a44a851eb78ed461b6051b09fdae0813"]
		}
	}
}]`

**Comment by filionf** [comment]
Made more progress today. I can reproduce using the code below (Node & Browser). Everything seems to be stored appropriately, and the problem would be in retrieving the information. I will try something and run the unit tests to see if I can fix it.

```
let db = new PouchDB('db');
let docs = [
  {
		"_id": "TEST",
		"_rev": "3-deleted",
		"_deleted": true,
		"_revisions": {
			"start": 3,
			"ids": ["deleted", "0a21b4bd4399b51e144a06b126031edc", "created"]
		}
	},
  {
		"_id": "TEST",
		"_rev": "3-0a21b4bd4399b51e144a06b126031edc",
		"_revisions": {
			"start": 3,
			"ids": ["0a21b4bd4399b51e144a06b126031edc", "updated", "created"]
		}
	}
];

resetDatabase().then(createDocs).then(logDoc, console.error);




function resetDatabase() {
  db = new PouchDB('db');
  return db.destroy().then(function() {
    db = new PouchDB('db');
  });
}

function createDocs() {
  let promise = Promise.resolve();
  for(let doc of docs) {
    promise = promise.then(updateDoc.bind(null, doc));
  }

  return promise;
}

function updateDoc(doc) {
  return db.bulkDocs([doc], { new_edits: false });
}

function logDoc() {
  return db.get('TEST', { revs: true, conflicts: true }).then(console.log);
}

```

**Comment by nolanlawson** [comment]
fixed in #5888
